camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/policy/GroupsChangePolicyOsgiTest.java

```diff
@@ -0,0 +1,472 @@
+                }
+            }).build());
+        }
+    }
+
+    /**
+     * Tests {@link GroupsChangePolicy#INITIALIZERS} of the member while joining the {@link DynamicGroup},
+     * {@link StaticSensor} in particular.
+     */
+    @Test
+    public void testAddInitializers() throws Exception {
+
+        // Member properties, these are expected to be resolved with DSL in the member context.
+        final String MEMBER_SENSOR_VALUE = "member-sensor-value";
+
+        // Group properties, these should not preempt member properties.
+        final String GROUP_SENSOR_VALUE = "group-sensor-value";
+
+        // Config names.
+        final String SENSOR_TO_COPY_FROM = "sensor-to-copy-from";
+        final String MEMBER_SENSOR_TO_CREATE = "sensor-to-create";
+
+        // Blueprint to test.
+        String yaml = Joiner.on("\n").join(
+                "services:",
+                "- type: " + DynamicGroup.class.getName(),
+                "  brooklyn.policies:",
+                "  - type: " + GroupsChangePolicy.class.getName(),
+                "    brooklyn.config:",
+                "      group: $brooklyn:self()",
+                "      member.initializers:",
+                "      - type: " + StaticSensor.class.getName(),
+                "        brooklyn.config:",
+                "          name: " + MEMBER_SENSOR_TO_CREATE,
+                "          target.type: string",
+                "          static.value: $brooklyn:attributeWhenReady(\"" + SENSOR_TO_COPY_FROM + "\")",
+                "- type: " + TestEntity.class.getName());
+
+        final Entity app = createStartWaitAndLogApplication(yaml);
+        Asserts.assertNotNull(app);
+
+        final DynamicGroup dynamicGroup = (DynamicGroup) Iterables.getFirst(app.getChildren(), null);
+        Asserts.assertNotNull(dynamicGroup);
+
+        final TestEntity member = (TestEntity) Iterables.getLast(app.getChildren());
+        Asserts.assertNotNull(member);
+
+        // Verify that member has joined.
+        dynamicGroup.setEntityFilter(EntityPredicates.idEqualTo(member.getId()));
+        Asserts.assertEquals(dynamicGroup.getMembers().size(), 1);
+
+        // Emmit dynamic attributes for both: group and its member.
+        executor.submit(() -> dynamicGroup.sensors().set(Sensors.newStringSensor(SENSOR_TO_COPY_FROM), GROUP_SENSOR_VALUE));
+        executor.submit(() -> member.sensors().set(Sensors.newStringSensor(SENSOR_TO_COPY_FROM), MEMBER_SENSOR_VALUE));
+
+        // Verify that location is added to the group member with expected member properties.
+        Asserts.eventually(() -> member.getAttribute(Sensors.newStringSensor(MEMBER_SENSOR_TO_CREATE)), MEMBER_SENSOR_VALUE::equals);
+    }
+
+    /**
+     * Tests {@link GroupsChangePolicy#INITIALIZERS} of the member while joining the {@link DynamicGroup},
+     * {@link CreatePasswordSensor} in particular.
+     */
+    @Test
+    public void testAddInitializers_CreatePasswordSensor() throws Exception {
```

Comment:
This is an original test, it has been rewritten for YAML test fixture.

Reply:
my point is that more than half of each test is the same.  so introducing a method to do that will make the relevant parts of each test easier to understand and to maintain.

maybe also worth it to add:

```
public void addClassNameType(Class<?> type, String itemType) {
        addCatalogItems(
                "brooklyn.catalog:",
                "  items:",
                "  - id: " + type.getName(),
                "    itemType: "+itemType,
                "    item:",
                "      type: " + type.getName());
}
```

Reply:
Yes, agree.

Reply:
Improved 6c19c45be964d66a89f8e6a8b339d2ba69674955